src/distilabel/llm/huggingface/inference_endpoints.py

```diff
@@ -121,10 +157,21 @@ def __init__(
         self.top_p = top_p
         self.typical_p = typical_p
 
-        self.inference_endpoint = get_inference_endpoint(
-            name=endpoint_name, namespace=endpoint_namespace, token=token
-        )
-        self.inference_endpoint.wait(timeout=30)
+        if is_severless_endpoint_available(model_id=endpoint_name_or_model_id):
+            logger.info("Using Serverless Inference Endpoint")
+            self.client = InferenceClient(model=endpoint_name_or_model_id, token=token)
+            self._model_name = endpoint_name_or_model_id
+        else:
+            logger.info("Using Dedicated Inference Endpoint")
+            inference_endpoint = get_inference_endpoint(
+                name=endpoint_name_or_model_id,
+                namespace=endpoint_namespace,
+                token=token,
+            )
+            inference_endpoint.wait(timeout=30)
```

Comment:
```suggestion
            inference_endpoint.resume().wait(timeout=30)
```
maybe resume before waiting for 30s?